source/core_http_client.c

```diff
@@ -1327,18 +1327,18 @@ static HTTPStatus_t addHeader( HTTPRequestHeaders_t * pRequestHeaders,
     /* Backtrack before trailing "\r\n" (HTTP header end) if it's already written.
      * Note that this method also writes trailing "\r\n" before returning.
      * The first condition prevents reading before start of the header. */
-    if( ( HTTP_HEADER_END_INDICATOR_LEN <= pRequestHeaders->headersLen ) &&
-        ( strncmp( ( char * ) pBufferCur - HTTP_HEADER_END_INDICATOR_LEN,
-                   HTTP_HEADER_END_INDICATOR, HTTP_HEADER_END_INDICATOR_LEN ) == 0 ) )
+    if( ( 4U <= pRequestHeaders->headersLen ) &&
+        ( strncmp( ( char * ) pBufferCur - 4U,
+                   "\r\n\r\n", 4U ) == 0 ) )
     {
-        backtrackHeaderLen -= HTTP_HEADER_LINE_SEPARATOR_LEN;
-        pBufferCur -= HTTP_HEADER_LINE_SEPARATOR_LEN;
+        backtrackHeaderLen -= 2U;
+        pBufferCur -= 2U;
     }
 
     /* Check if there is enough space in buffer for additional header. */
-    toAddLen = fieldLen + HTTP_HEADER_FIELD_SEPARATOR_LEN + valueLen +
-               HTTP_HEADER_LINE_SEPARATOR_LEN +
-               HTTP_HEADER_LINE_SEPARATOR_LEN;
```

Comment:
IMO `HTTP_HEADER_LINE_SEPERATOR_LEN` is less magical than `2U` sprinkled around the file.

Reply:
I switched this to use a CONST_STRLEN() preprocessor macro to clarify things.